File: dotnet/Samples/LinguisticOptions/EdgeLinguisticOptionsService.cs
﻿using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.LinguisticOptions.Edge.Request;
using LanguageWeaver.Sdk.LinguisticOptions.Edge.Result;

namespace LanguageWeaver.Sdk.Samples.LinguisticOptions;

public class EdgeLinguisticOptionsService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();

        EdgeGetLinguisticOptionsRequest edgeGetLinguisticOptionsRequest = new EdgeGetLinguisticOptionsRequest()
        {
            SourceLanguageId = "chi",
            TargetLanguageId = "eng",
            Model = "Generic"
//            Platform = "SRV",
//            Technology = "TNMV"
        };

        EdgeLinguisticOptionsResult edgeLinguisticOptionsResult =
            lwClient.GetEdgeLinguisticOptions(edgeGetLinguisticOptionsRequest);
        // handle result
    }
}

File: javascript/backend/samples/linguisticOptions/EdgeLinguisticOptionsService.js
const {EdgeLanguageWeaverClient, EdgeGetLinguisticOptionsRequest} = require("@language-weaver/lw-sdk-js");

const getLinguisticOptions = async () => {
    try {
        const edgeLanguageWeaverClient = await new EdgeLanguageWeaverClient().build();
        const request = new EdgeGetLinguisticOptionsRequest();
        request.sourceLanguageId = "chi";
        request.targetLanguageId = "eng";
        request.model = "Generic";
        // request.platform = "SRV";
        // request.technology = "TNMV";
        const linguisticOptionsResult = await edgeLanguageWeaverClient.getEdgeLinguisticOptions(request);
        console.log(linguisticOptionsResult.linguisticOptions);
    } catch (e) {
        console.log(e);
    }
}

getLinguisticOptions();